src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const dateParts = date.split(separator);
+  const newDate = ['', '', ''];
+
+  const formatMap = {
+    DD: 'день',
+    MM: 'місяць',
+    YYYY: 'рік',
+    YY: 'рік',
+  };
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const format = fromFormat[i];
+    const partType = formatMap[format];
+    const partValue = dateParts[i];
+
+    let newIndex = -1;
+
+    for (let j = 0; j < toFormat.length; j++) {
+      if (formatMap[toFormat[j]] === partType) {
+        newIndex = j;
+        break;
+      }
+    }
+
+    if (format === 'YYYY' && toFormat[newIndex] === 'YY') {
+      newDate[newIndex] = partValue.slice(-2);
+    } else if (format === 'YY' && toFormat[newIndex] === 'YYYY') {
+      newDate[newIndex] = (+partValue < 30 ? '20' : '19') + partValue;
```

Comment:
The conversion from 'YY' to 'YYYY' assumes that all years below '30' belong to the 21st century, and all others to the 20th century. This is a reasonable assumption for most recent dates but may not hold true universally. If your application needs to handle historical dates or future dates beyond 2029, additional logic will be required.